enhancements/monitoring/alert-overrides.md

```diff
@@ -0,0 +1,280 @@
+  metrics from components in the `openshift-*` namespaces.
+
+### API Extensions
+
+None, but this is an open question.  Is there good reason to make this a CRD
+rather than a simple `ConfigMap`?  If so, it may make sense to start with a
+`ConfigMap` in tech-preview for the 4.10 release, then move to a CRD later.
+
+### Implementation Details/Notes/Constraints
+
+TBD
+
+### Risks and Mitigations
+
+The primary risk is that this provides users with a method to permanently
+modify, including to completely drop, alerting rules, which could lead to
+unintentionally missed alerts in the case of user error.  Careful consideration
+should be given to UX.  Information on which overridden or dropped alerts are
+firing is still available in the Prometheus API.  Perhaps this can be leveraged
+and surfaced in an unobtrusive way.
+
+## Design Details
+
+### Open Questions
+
+- Should we allow targeting by labels other than name and severity?  E.g. should
+  a user be able to target a critical alert only in a single namespace?
+- Should we introduce a CRD for this, or is a simple `ConfigMap` enough?
+
+### Test Plan
+
+This should be fairly straight forward to both unit and e2e test.
+
+### Graduation Criteria
+
+Plan to release as Tech Preview initially.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+Upgrade and downgrade are not expected to present a significant challenge.  The
+overridden rules are reconciled against those shipped by OpenShift in real time,
+so any updates are applied as needed.  On downgrade to a version not supporting
+the feature, the newly generated `PrometheusRule` resource and the `Secret`
+containing the relabeling configs can simply be deleted.
+
+The biggest concern is removal, renaming, or change in severity of OpenShift
+shipped rules as this may prevent a previously overridden rule from being found.
+
+### Version Skew Strategy
+
+This shouldn't be an issue as rules are reconciled against OpenShift rules and
+regenerated as necessary as changes occur.
+
+### Operational Aspects of API Extensions
+
+N/A -- This needs to be addressed if it's decided that a CRD is appropriate.
+
+#### Failure Modes
+
+TBD
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+- Initial Proof-of-Concept:
+  https://github.com/openshift/cluster-monitoring-operator/pull/1473
+
+## Drawbacks
+
+TBD
+
+## Alternatives
```

Comment:
It would be useful to describe here why a ConfigMap is being used, instead of defining a new CRD for a regular API.

Reply:
Yeah, to be honest I'm not sure there's a reason other than that's what we prototyped with. It's called out as an open question. If the consensus is to just go with a CRD from the start, we can do that.

Reply:
We ended up deciding a CRD is best. I've added some detail about the proposed structure.